src/Span.ts

```diff
@@ -364,17 +356,31 @@ class Span implements Partial<ISpan> {
    */
 
   private async recordMessage(message: TypeOfMessage, type: TypeEventTypeAction['type'], action?: TypeEventTypeAction['action'], state?: EventStateMetadata) {
-    if (this.isFinished) throw new Error('span finished. no further actions allowed')
+    if (this.isFinished) {
+      throw new Error('span finished. no further actions allowed')
+    }
+
     let newEnvelope = this.createEventMessage(message, type, action, state)
-    let logResult
     let key = <RecorderKeys>`${type}Recorder`
-    if (this.recorders[key]) logResult = await this.recorders[key]!.record(newEnvelope, Config.EVENT_LOGGER_SIDECAR_WITH_LOGGER)
-    else logResult = await this.recorders.defaultRecorder.record(newEnvelope, Config.EVENT_LOGGER_SIDECAR_WITH_LOGGER)
-    if (LogResponseStatus.accepted == logResult.status) {
-      return logResult
-    } else {
+
+    let recorder = this.recorders.defaultRecorder
+    if (this.recorders[key]) {
+      recorder = this.recorders[key]!
+    }
+
+    if (Config.ASYNC) {
+      //Don't wait for .record() to resolve, return straight away
+      recorder.record(newEnvelope, Config.EVENT_LOGGER_SIDECAR_WITH_LOGGER)
```

Comment:
This is where the magic lies. If we want async logging, then don't wait for `recorder.record()` to return.